homeassistant/components/shelly/service.py

```diff
@@ -0,0 +1,36 @@
+"""Services for the Shelly integration."""
+from __future__ import annotations
+
+from homeassistant.const import ATTR_AREA_ID, ATTR_DEVICE_ID
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.service import async_extract_config_entry_ids
+
+from .const import ATTR_BETA, BLOCK, DATA_CONFIG_ENTRY, DOMAIN, RPC, SERVICE_OTA_UPDATE
+from .utils import get_device_entry_gen
+
+
+async def async_services_setup(hass: HomeAssistant) -> None:
+    """Set up services."""
+
+    async def async_service_ota_update(call: ServiceCall) -> None:
+        """Trigger OTA update."""
+        if not (call.data.get(ATTR_DEVICE_ID) or call.data.get(ATTR_AREA_ID)):
+            raise HomeAssistantError("No target selected for OTA update")
+
+        beta_channel = bool(call.data.get(ATTR_BETA))
+
+        entry_ids = await async_extract_config_entry_ids(hass, call)
+        for entry_id in entry_ids:
+            entry = hass.config_entries.async_get_entry(entry_id)
+            if not (entry and entry.domain == DOMAIN):
+                continue
```

Comment:
Would it be possible to add a test to cover this line?

Reply:
did already think about it, but struggle a bit how to setup a valid "fake" device from any other domain as the shelly domain itself during the test 🤔

Reply:
not necessary anymore